### DISCUSSION TITLE
fixing wrong Content-Base ip?

### DISCUSSION THREAD

**Discussion by emiltin** [bluenviron/mediamtx#897] [Questions]
Hi,
Thank you for an awesome project!

We're proxying cameras, and some work. But others don't, apparently because the cameras is behind a router with NAT, and the cameras insists on placing it's local ip in the Content-Base field of the response to the DESCRIBE request.

As far as I can tell, rtsp-simple-server is then using this ip when trying playing the stream, which fails because it's local ip behind the router.

The correct ip is in fact in the rtps-simple-server config for the camera.

config:
```
paths:
  AG0605-1:
    source: rtsp://12.12.12.12:1333/h264
    sourceProtocol: tcp
```

rtps-simple-server log:
```
2022/03/24 10:23:40 DEB [path AG0605-1] [rtsp source] c->s DESCRIBE rtsp://12.12.12.12:1333/h264 RTSP/1.0
Accept: application/sdp
Authorization: Basic xxxxxxxxxxxxxxxxxxx=
CSeq: 3
User-Agent: gortsplib

2022/03/24 10:23:40 DEB [path AG0605-1] [rtsp source] s->c RTSP/1.0 200 OK
CSeq: 3
Content-Base: rtsp://192.168.0.250:1333/h264/       <---- this ip is the problem
Content-Length: 716
Content-Type: application/sdp
Date: Thu, Mar 24 2022 10:23:40 GMT
...
```

The ip12.12.12.12 in the DESCRIBE is correct (although i've changed it here), and matches the config.
The ip 192.168.0.250 in the response is wrong (due to a bug in the camera I guess).

Is there a way to make rtsp-simple-server use the ip from the config when playing the stream, instead of from the Content-Base?

**Comment by aler9** [comment]
Thanks for the report, please open an issue and provide a network dump (or server log) in order to allow us to replicate the bug. You're right, the host and port provided by Content-Base are used in subsequent SETUP requests. Since a fix may break compatibility with other cameras, a network dump (or server log) is needed in order to correctly debug the error message (which you didn't provide). Thanks.

**Comment by aler9** [comment]
I just encountered this problem, it turned out that when there was a UDP->TCP protocol switch, the IP of the camera was overridden with the one contained into Content-Base. This is fixed in v0.18.2.
